File: src/main/java/recipestore/db/triplestore/CommonModule.java
package recipestore.db.triplestore;

import com.google.inject.AbstractModule;
import com.google.inject.name.Names;
import recipestore.ResourceLoader;

public class CommonModule extends AbstractModule {

    String GRAPHFRAME_DIR = "graphframe_dir";

    protected final String graphDir = String.format("file://%s", ResourceLoader.get.apply(ResourceLoader.Resource.graph,
            GRAPHFRAME_DIR).get());

    @Override
    protected void configure() {
        bind(String.class).annotatedWith(Names.named("graphDirectory"))
                .toInstance(graphDir);
    }
}

File: src/main/scala/recipestore/CommonModule.scala
package recipestore

import com.google.inject.AbstractModule
import com.google.inject.name.Names
import net.codingwell.scalaguice.ScalaModule

class CommonModule extends AbstractModule with ScalaModule {
  private[recipestore] val GRAPHFRAME_DIR: String = "graphframe_dir"
  final protected val graphDir: String = String.format("file://%s", ResourceLoader.apply(AppResource.GraphResource, GRAPHFRAME_DIR).get)

  override protected def configure() {
    bind(classOf[String]).annotatedWith(Names.named("graphDirectory")).toInstance(graphDir)
  }
}